utils/regr/regr_intercept.cpp

```diff
@@ -154,7 +154,7 @@ mcsv1_UDAF::ReturnCode regr_intercept::evaluate(mcsv1Context* context, static_an
         long double sumxy = data->sumxy;
         long double numerator = sumy * sumx2 - sumx * sumxy;
         long double var_pop = (N * sumx2) - (sumx * sumx);
-        if (var_pop != 0)
+        if (var_pop != 0 && var_pop != -0)
         {
```

Comment:
This change is meaningless I am afraid:
0 and -0 are integer (not DOUBLE!) literals and both are equal to INT(0), and then both are converted to DOUBLE to comparison.

So possibly you meant:

`if (var_pop != 0e0 && var_pop != -0e0)`

But this is not really needed because  0e0 and -0e0 are equal to each other although they are not binary equal.
This program proves:

```
#include <stdio.h>
#include <cstring>

int main()
{
  double a= -0e0;
  double b= +0e0;
  printf("memcmp=%d eq=%d\n", memcmp(&a, &b, sizeof(a)), a == b);
  return 0;
}
```
And the output is:
```
memcmp=128 eq=1
```

I guess the old version of this line was good enough.

Reply:
Sometimes it appears to not compare properly. I don't know if this is an artifact of other bits being different or perhaps a float number so small in abs value that it prints as -0. Whatever the case, we can use the fact that var_pop should never be < 0.

Reply:
After some research, it turns out var_pop cannot be negative, but covar_pop may be negative. The changes assume <0 should be 0 for var_pop and any negative values for covar_pop remain as calculated. This does allow for -0 to be printed in the answer for some functions.